File: CIVITASRUBY/lib/salir_carcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  
  require_relative 'sorpresa'
  class Salir_carcel < Sorpresa
    
    def initialize(tablero)
      super(tablero ,-1, "salir carcel" , nil)
    end
    
      def aplicar_a_jugador(actual,todos)
      if jugador_correcto(actual,todos)
        informe(actual,todos)
        la_tienen = false
        
        todos.each do |j|
          if j.tiene_salvoconducto()
            la_tienen = true
          end
        end
        

        
        if la_tienen == false
          todos[actual].obtener_salvoconducto(self)
          salir_del_mazo()
        end
      end
    end
    
  end
end

File: CIVITAS/src/civitas/SalirCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Usuario
 */
public class SalirCarcel extends Sorpresa {

    SalirCarcel(Tablero tablero){
        super(tablero ,-1, "salir carcel" , null);
    }    
    
    
    @Override
     void aplicarAJugador(int actual  ,ArrayList<Jugador> todos){
        if(this.jugadorCorrecto(actual, todos)){
            this.informe(actual, todos);
            Boolean latienen = false;
            for(Jugador j : todos){
                 if( j.tieneSalvoconducto()){
                     latienen = true;
                 }
            }

            if(!latienen){
                todos.get(actual).obtenerSalvoconducto(this);
                this.salirDelMazo();
            }

        }

    }
}
